### PR TITLE
deploy: Add k8s YAML

### PR DESCRIPTION
**Pull request by Jongy**
## Description
Add a basic template YAML for k8s deployment of gProfiler.

## Motivation and Context
Allow deploying gProfiler easily over k8s.

## How Has This Been Tested?
On my local k8s cluster.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:
- [x] My code follows the code style of this project.
- [x] I have updated the relevant documentation.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Jongy** [comment]
Not sure what are these errors now :/

**Comment by leonid-granulate** [comment]
> Not sure what are these errors now :/

See https://github.com/Granulate/gprofiler/pull/34

File: deploy/k8s/gprofiler.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        - name: gprofiler
+          securityContext:
+            privileged: true
+          resources:
+            requests:
+              cpu: 100m
+              memory: 256m
+            limits:
+              cpu: 512m
+              memory: 1Gi
```
[the remark this answers is not in the document]
Comment (Jongy) [reply deploy/k8s/gprofiler.yaml:37]: It depends on many variables:
* how many different binaries are running (will affect `perf` when it's copying files to its buildid cache)
* how many Python processes are running (will affect linearly the number of `py-spy`s)
* ...

I think that we can go with those limits and amend them later if they cause issues anywhere. Additionally we can ensure our tools are constrained to these limits (avoid running too many `py-spy`s, prevent `perf` from eating up memory, ...)

File: deploy/k8s/gprofiler.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      containers:
+        - name: gprofiler
+          securityContext:
+            privileged: true
+          resources:
+            requests:
+              cpu: 100m
+              memory: 256m
+            limits:
+              cpu: 512m
```
Comment (leonid-granulate) [inline deploy/k8s/gprofiler.yaml:36]: The cpu value of 512m is not make sense here, 1000m is one core so 500m is a half core (not powers of 2).
https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#cpu-units

Comment (Jongy) [reply deploy/k8s/gprofiler.yaml:36]: You are absolutely right

**Review by leonid-granulate — APPROVED** [review]

**Pull request merged**